run_diff.go

```diff
@@ -147,10 +153,26 @@ func diffResults(before map[string][]int, after map[string][]int) map[string][]i
 		}
 	}
 	for test, resultsAfter := range after {
+		if paths != nil && !anyPathMatches(paths, test) {
+			continue
+		}
+
 		if _, ok := before[test]; !ok {
 			// Missing? Then N / N tests are 'different'
 			diff[test] = []int{resultsAfter[1], resultsAfter[1]}
 		}
 	}
 	return diff
 }
+
+func anyPathMatches(paths mapset.Set, testPath string) bool {
+	if paths == nil {
+		return true
```

Comment:
If nil is a way to represent an empty set, then this is the opposite of what I'd expect. Maybe assert that it's not null and handle it outside?

Reply:
nil is 'no filter set', which is not the same as an empty set.